File: src/vowel.java
public class vowel {
    public static void main(String[] args) {
        String input = "abide";
        System.out.println(isValid(input) ? "Valid" : "Invalid");
    }

    static boolean isValid(String word) {
        if (word.length() < 3) return false;
        String vowels = "aeiouAEIOU";
        if (!vowels.contains("" + word.charAt(0)) || !vowels.contains("" + word.charAt(word.length() - 1)))
            return false;

        int count = 0;
        for (int i = 1; i < word.length() - 1; i++) {
            if (vowels.contains("" + word.charAt(i))) count++;
        }

        return count == 2;
    }
}

File: vowel.py
def is_valid_puzzle_word(word):
    vowels = 'aeiouAEIOU'
    if len(word) < 3:
        return "Invalid"  # Not enough letters to have 2 inside vowels

    if word[0] in vowels and word[-1] in vowels:
        # Count vowels excluding first and last
        inside_vowels = sum(1 for ch in word[1:-1] if ch in vowels)
        return "Valid" if inside_vowels == 2 else "Invalid"
    return "Invalid"

# Example
print(is_valid_puzzle_word("abide"))  # Output: Valid
